"zar import" panic starting at commit 0382a0f

I was running through the steps in the [`zar` README](https://github.com/brimsec/zq/blob/master/cmd/zar/README.md) on `master` with intent to verify #1262, but I found a `zar import` panic instead. I did a binary search and confirmed the panic started with `zar` commit `0382a0f`, which is associated with #1345.

```
$ rm -rf $ZAR_ROOT ; zq zng/*.gz | zar import -s 25MB -
panic: runtime error: slice bounds out of range [:52] with capacity 27

goroutine 1 [running]:
github.com/brimsec/zq/zcode.(*Iter).Next(0xc0001334b0, 0xc0155bc1e1, 0x39, 0x54, 0xc000044500, 0x0, 0x0)
	/Users/phil/work/zq/zcode/iter.go:30 +0x242
github.com/brimsec/zq/zng.(*Record).Slice(0xc0155be000, 0x1, 0x2, 0x201, 0x90000c000024800, 0xc000133548, 0x0)
	/Users/phil/work/zq/zng/recordval.go:200 +0x8d
github.com/brimsec/zq/zng.(*Record).Value(0xc0155be000, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/zng/recordval.go:211 +0x58
github.com/brimsec/zq/zng.(*Record).Access(0xc0155be000, 0x18799d6, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x57, 0x200)
	/Users/phil/work/zq/zng/recordval.go:239 +0x8f
github.com/brimsec/zq/zng.(*Record).AccessTime(0xc0155be000, 0x18799d6, 0x2, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/zng/recordval.go:305 +0x4c
github.com/brimsec/zq/zng.(*Record).Ts(...)
	/Users/phil/work/zq/zng/recordval.go:327
github.com/brimsec/zq/archive.(*importDriver).writeOne(0xc000726ea0, 0xc0155be000, 0xc0155be000, 0x0)
	/Users/phil/work/zq/archive/import.go:84 +0x40d
github.com/brimsec/zq/archive.(*importDriver).Write(0xc000726ea0, 0x0, 0x1ab4ca0, 0xc00ea2e280, 0x0, 0x0)
	/Users/phil/work/zq/archive/import.go:171 +0x92
github.com/brimsec/zq/driver.runMux(0xc000032280, 0x1ab0f20, 0xc000726ea0, 0x0, 0xc000032820, 0x1aa7860)
	/Users/phil/work/zq/driver/driver.go:65 +0x253
github.com/brimsec/zq/driver.MultiRun(0x1ab0fe0, 0xc000249890, 0x1ab0f20, 0xc000726ea0, 0x1a9ed40, 0xc000323ce0, 0xc000032820, 0x1aa7860, 0xc000323e00, 0x0, ...)
	/Users/phil/work/zq/driver/driver.go:36 +0x18d
github.com/brimsec/zq/driver.Run(0x1ab0fe0, 0xc000249890, 0x1ab0f20, 0xc000726ea0, 0x1a9ed40, 0xc000323ce0, 0xc000032820, 0x1a9f0c0, 0xc000249860, 0x0, ...)
	/Users/phil/work/zq/driver/driver.go:25 +0x26b
github.com/brimsec/zq/archive.Import(0x1ab0fe0, 0xc000249890, 0xc0000d1900, 0xc000032820, 0x1a9f0c0, 0xc000249860, 0x4, 0x1)
	/Users/phil/work/zq/archive/import.go:206 +0x191
github.com/brimsec/zq/cmd/zar/import.(*Command).Run(0xc00027d9a0, 0xc0000320e0, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/work/zq/cmd/zar/import/command.go:104 +0x547
github.com/mccanne/charm.(*instance).run(0xc00000fd40, 0xc0000320c0, 0x3, 0x3, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).Exec(0x1fc98e0, 0x1a9ef40, 0xc000318340, 0xc0000320c0, 0x3, 0x3, 0x4, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:67 +0x92
github.com/mccanne/charm.(*instance).run(0xc00000fd20, 0xc0000320b0, 0x4, 0x4, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:61 +0x41d
github.com/mccanne/charm.(*Spec).ExecRoot(0x1fc9ca0, 0xc0000320b0, 0x4, 0x4, 0xc00029ff78, 0x100757f, 0xc00009a058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zar/main.go:20 +0x78
```

--- a/proc/spill/peeker.go
+++ b/proc/spill/peeker.go
@@ -37,6 +37,9 @@ func newPeeker(filename string, recs []*zng.Record, zctx *resolver.Context) (*pe
 // do its heap management a bit more easily.
 func (p *peeker) read() (*zng.Record, bool, error) {
 	rec := p.nextRecord
+	if rec != nil {
+		rec = rec.Keep()
+	}
 	var err error
 	p.nextRecord, err = p.Read()
 	eof := p.nextRecord == nil && err == nil

